src/components/dialogs/line-split-with-voltage-level-dialog.js

```diff
@@ -201,9 +203,14 @@ const LineSplitWithVoltageLevelDialog = ({
         });
 
     useEffect(() => {
+        const vlId =
+            typeof voltageLevelOrId === 'string'
+                ? voltageLevelOrId
+                : voltageLevelOrId?.id;
         if (
-            newVoltageLevelRef.current !== null &&
-            voltageLevelOrId !== voltageLevelOrIdRef.current
+            newVoltageLevelRef.current &&
+            voltageLevelOrIdRef.current &&
+            vlId !== newVoltageLevelRef.current.equipmentId
         ) {
             voltageLevelOrIdRef.current = voltageLevelOrId;
             setNewVoltageLevel(null);
```

Comment:
Maybe change the following line to `newVoltageLevelRef.current = null;`, and remove dependency to newVoltageLevel since it might be dangerous to use setNewVoltageLevel and newVoltageLevel in a useEffect.

Reply:
Done